server/src/daemon/session.rs

```diff
@@ -0,0 +1,56 @@
+// org.freedesktop.Secret.Session
+
+use std::sync::{Arc, Mutex};
+
+use oo7::Key;
+use zbus::{fdo, interface, zvariant};
+use zvariant::{ObjectPath, OwnedObjectPath};
+
+use super::service_manager::ServiceManager;
+
+const SECRET_SESSION_PREFIX: &str = "/org/freedesktop/secrets/session/";
+
+#[derive(Debug, Clone)]
+pub struct Session {
+    aes_key: Arc<Option<Key>>,
+    manager: Arc<Mutex<ServiceManager>>,
+    pub path: OwnedObjectPath,
+}
+
+#[interface(name = "org.freedesktop.Secret.Service")]
+impl Session {
+    pub async fn close(
+        &mut self,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> fdo::Result<()> {
+        self.manager.lock().unwrap().remove_session(self.path());
+        object_server.remove::<Self, _>(&self.path).await?;
+        Ok(())
+    }
+}
+
+impl Session {
+    pub fn new(
+        aes_key: Option<Key>,
+        manager: Arc<Mutex<ServiceManager>>,
+        sessions_counter: i32,
+    ) -> Self {
+        Self {
+            aes_key: Arc::new(aes_key),
+            path: OwnedObjectPath::try_from(format!(
+                "{}{}{}",
+                SECRET_SESSION_PREFIX, "s", sessions_counter
```

Comment:
What is `s` here?

Reply:
`s` comes from [here](https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/master/daemon/dbus/gkd-secret-session.c?ref_type=heads#L313). As in `/org/freedesktop/secrets/session/s1`

Reply:
Then there is no point in not writing it directly in `{}s{}`